### PR TITLE
Update Chained Exception test from #965

### PR DESCRIPTION
**Pull request by duytnguyendtn**
### Description

Enhances one of the display tests from #965 to check for the explicit error, rather than the outer, generic, chained, user-friendly error

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Is this really what you want? Or you rather raise a straight up `NotImplementedError`?

**Comment by duytnguyendtn** [comment]
@pllim RE https://github.com/spacetelescope/jdaviz/pull/965#issuecomment-1175501315: I'm actually explicitly trying to wrap all the errors with the RuntimeError. Specifically, I wanted to explicitly wrap all the errors with the RuntimeError message to "tell the story" of why things broke. It's probably obvious for us as SEs when an exception raises when running `show()` that there was an error displaying Jdaviz, but I want to try to help along the super novice by adding some "user friendly padding", which is what that RuntimeError is

**Comment by duytnguyendtn** [comment]
And FWIW, this isn't really for the NotImplemented Error, this should also be for the ValueError as well for the unrecognized arg

**Comment by duytnguyendtn** [comment]
So really it should look like https://github.com/spacetelescope/jdaviz/pull/1454/commits/1b8605e32c7f3548f0645ca2c4eab138c6ecfd99

**Review by pllim — CHANGES_REQUESTED** [review]
I think you can write the rest without try/except like this:

```python
with pytest.raises(RuntimeError, match='Error in displaying Jdaviz') as err:
    ...
assert ....
```

**Comment by duytnguyendtn** [comment]
So like that https://github.com/spacetelescope/jdaviz/pull/1454/commits/6d84917c4e3255efe0e9de5565e8312416aa2e93?

File: jdaviz/tests/test_show.py [hunk 1]
```diff
@@ -2,10 +2,12 @@
 
 
 def test_show_popout(specviz_helper):
-    with pytest.raises(RuntimeError, match='Error in displaying Jdaviz'):
+    with pytest.raises(RuntimeError, match='Error in displaying Jdaviz') as err:
         specviz_helper.show(loc="popout")
+        assert isinstance(err.__cause__, NotImplementedError)
```
Comment (pllim) [inline jdaviz/tests/test_show.py:7]: Yes, but let's do that outside of the context manager as pytest recommended in https://docs.pytest.org/en/stable/how-to/assert.html#assertions-about-expected-exceptions 😸 

```suggestion
    assert isinstance(err.__cause__, NotImplementedError)
```

File: jdaviz/tests/test_show.py [hunk 2]
```diff
@@ -2,10 +2,12 @@
...
 
 
 def test_show_invalid_mode(specviz_helper):
-    with pytest.raises(RuntimeError, match='Error in displaying Jdaviz'):
+    with pytest.raises(RuntimeError, match='Error in displaying Jdaviz') as err:
         specviz_helper.show(loc="thisisnotavaliddisplaymode")
+        assert isinstance(err.__cause__, ValueError)
```
Comment (pllim) [inline jdaviz/tests/test_show.py:13]:
```suggestion
    assert isinstance(err.__cause__, ValueError)
```

**Comment by duytnguyendtn** [comment]
OH lol clearly my C brain was engaged because I was thinking "Oh, if the assert isn't under the context manager, how can it access the exception defined inside?? 😆

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

p.s. Didn't know there is a `__cause__`. TIL!

**Comment by duytnguyendtn** [comment]
Whoops, something broke.... one sec

**Comment by pllim** [comment]
Hmm... Might need a slightly different syntax to access the cause since pytest exception object isn't pure Python object. 😬

**Comment by duytnguyendtn** [comment]
Found it! https://stackoverflow.com/questions/70499076/how-to-check-exception-cause-using-pytest-raises
```
err.value.__cause__
```

**Comment by duytnguyendtn** [comment]
`trivial` = 1 approval, so I'll merge once the tests pass (though if you catch it before I do, feel free Pey-Lian!)

**Comment by pllim** [comment]
Please use the "Squash and Merge" button. Thanks!

**Pull request merged**